cpp/src/slice2swift/SwiftUtil.cpp

```diff
@@ -15,28 +15,68 @@ using namespace IceInternal;
 
 namespace
 {
-    // TODO: fix this to emit double-ticks instead of single-ticks once we've fixed all the links.
-    // TODO: this is temporary and will be replaced when we add 'swift:identifier' support.
-    string swiftLinkFormatter(const string& rawLink, const ContainedPtr&, const SyntaxTreeBasePtr&)
+    string swiftLinkFormatter(const string& rawLink, const ContainedPtr& source, const SyntaxTreeBasePtr& target)
     {
-        string result = "`";
+        string result = "``";
 
-        auto hashPos = rawLink.find('#');
-        if (hashPos != string::npos)
+        if (auto builtin = dynamic_pointer_cast<Builtin>(target))
         {
-            if (hashPos != 0)
+            result += Slice::Swift::typeToString(builtin, source, false);
+        }
+        else if (auto contained = dynamic_pointer_cast<Contained>(target))
+        {
+            string nameSuffix;
+
+            // If the link is to a data member or an enumerator, we store its name in `nameSuffix`, then navigate up to
+            // its parent. Due to the flat nature of Swift, we always need to work up to something defined in a module.
+            if (dynamic_pointer_cast<DataMember>(contained) || dynamic_pointer_cast<Enumerator>(contained))
             {
-                result += rawLink.substr(0, hashPos);
-                result += "/";
+                nameSuffix = "/" + contained->mappedName();
+
+                // Move up a level to the container type.
+                contained = dynamic_pointer_cast<Contained>(contained->container());
+                assert(contained);
             }
-            result += rawLink.substr(hashPos + 1);
+
+            // If the link is to an operation, DocC requires you to list the operation's labels in the link signature.
+            if (auto operation = dynamic_pointer_cast<Operation>(contained))
+            {
+                nameSuffix = "/" + operation->mappedName() + "(";
+                const ParameterList inParams = operation->inParameters();
+                for (const auto& param : inParams)
+                {
+                    const string paramLabel = (inParams.size() == 1 ? "_" : param->mappedName());
+                    nameSuffix += paramLabel + ":";
+                }
+                nameSuffix += "context:)";
+
+                // Move up a level to the container type.
+                contained = dynamic_pointer_cast<Contained>(contained->container());
+                assert(contained);
+            }
+
+            // If the link was to an interface definition, we need to switch to the corresponding declaration.
+            // The code-gen considers `Def` the servant type, and `Decl` the proxy type. We want to link to the proxy.
+            if (auto interfaceDef = dynamic_pointer_cast<InterfaceDef>(contained))
```

Comment:
is this block also used for operations, after they move up? If it is not an `else if` would make the intention clear, if it is the comment should make it clear.